internal/compiler-bridge/src/main/scala/xsbt/DelegatingReporter.scala

```diff
@@ -109,8 +109,24 @@ private object DelegatingReporter {
       val endOffset = if (pos.isRange) Some(pos.end) else None
       val startLine = if (pos.isRange) Some(lineOf(pos.start)) else None
       val startColumn = if (pos.isRange) Some(columnOf(pos.start)) else None
-      val endLine = if (pos.isRange) Some(lineOf(pos.end)) else None
-      val endColumn = if (pos.isRange) Some(columnOf(pos.end)) else None
+      val endLine =
+        if (pos.isRange)
+          try {
+            Some(lineOf(pos.end))
+          } catch {
+            // work around for https://github.com/scala/bug/issues/11865 by falling back to start pos
+            case _: ArrayIndexOutOfBoundsException =>
+              startLine
+          } else None
+      val endColumn =
+        if (pos.isRange)
+          try {
+            Some(columnOf(pos.end))
+          } catch {
+            // work around for https://github.com/scala/bug/issues/11865 by falling back to start pos
+            case _: ArrayIndexOutOfBoundsException =>
+              startColumn
+          } else None
```

Comment:
You can trim this down with:

`Try(columnOf(pos.end)).recover { case _: ArrayIndexOutOfBoundsException => startColumn }.toOption`

Or, if you prefer:

```scala
import scala.util.control.Exception.catching

private val aioobeCatcher = catching(classOf[ArrayIndexOutOfBoundsException])

Some(aioobeCatcher.withApply(_ => startColumn)(columnOf(pos.end)))
```

Reply:
Maybe it's premature optimization, but I feel like delegating reporter should avoid allocation?

Reply:
Yep, that sounds like premature optimization to me.